libraries/chain/controller.cpp

```diff
@@ -1988,6 +1984,7 @@ struct controller_impl {
          ilog( "chain database started with hash: ${hash}", ("hash", calculate_integrity_hash()) );
       okay_to_print_integrity_hash_on_stop = true;
 
+      replaying = true;
       replay( startup ); // replay any irreversible and reversible blocks ahead of current head
 
       if( check_shutdown() ) return;
```

Comment:
Probably no need to set it back to false as node is going to shut down.

Reply:
Right, any exception or if we shutdown there is no need to reset it. Although, maybe better to be pedantic and and put it in a scope exit to reset.